src/Data/List/Lazy/Types.purs

```diff
@@ -105,6 +107,12 @@ instance functorList :: Functor List where
     go Nil = Nil
     go (Cons x xs') = Cons (f x) (f <$> xs')
 
+instance functorWithIndexList :: FunctorWithIndex Int List where
+  mapWithIndex f xs = List (go 0 <$> unwrap xs)
+    where
+    go _ Nil = Nil
+    go i (Cons x xs') = Cons (f i x) (f (i + 1) <$> xs')
```

Comment:
Shouldn't there be a recursive call to `go` here?

Maybe it's better to try to write this as a `foldr`?

Reply:
Ah, indeed. I think I was running my tests locally on the wrong branch by accident.

Reply:
Actually I think this is a different mistake I've made, besides the one breaking the test above. I'll go through and add some more thorough testing wherever it makes sense.